src/braket/tasks/local_quantum_task.py

```diff
@@ -30,38 +34,161 @@ class LocalQuantumTask(QuantumTask):
 
     def __init__(
         self,
-        result: Union[
-            GateModelQuantumTaskResult, AnnealingQuantumTaskResult, PhotonicModelQuantumTaskResult
-        ],
+        result: Optional[
+            Union[
+                GateModelQuantumTaskResult,
+                AnnealingQuantumTaskResult,
+                PhotonicModelQuantumTaskResult,
+            ]
+        ] = None,
     ):
-        self._id = result.task_metadata.id
         self._result = result
+        self._loop = asyncio.new_event_loop()
+
+    @staticmethod
+    def create(
+        execute_manager: QuantumExecuteManager,
+        *args,
+        **kwargs,
+    ) -> "LocalQuantumTask":
+        """LocalQuantumTask factory method that serializes a quantum task specification
+        (either a quantum circuit or problem), computes the result,
+        and returns back an LocalQuantumTask tracking the execution.
+
+        Args:
+            execute_manager (QuantumExecuteManager):  Execute Manager
+
+        Returns:
+            : LocalQuantumTask tracking the task execution on the device.
+        """
+        task = LocalQuantumTask()
+        task._execute_manager = execute_manager
+        task.async_result()
+        return task
 
-    @property
     def id(self) -> str:
-        return str(self._id)
+        raise NotImplementedError("Id is not generated for LocalQuantumTask")
 
     def cancel(self) -> None:
         """Cancel the quantum task."""
-        raise NotImplementedError("Cannot cancel completed local task")
+        if hasattr(self, "_execute_manager"):
+            return self._execute_manager.cancel()
+        raise NotImplementedError("LocalQuantumTask does not support cancelling")
```

Comment:
Instead of using `hasattr`, you can just do an `if self._execute_manager` check, and set it to `None` if not supplied

Reply:
`getattr(self, "_execute_manager", None)`?

Reply:
My intention is for the user to define execute manager.

I have has `hasattr(self, "_execute_manager"` in place for old test cases where we pass in result directly to LocalQuantumTask.